source/uri/Endpoint.ooc

```diff
@@ -0,0 +1,45 @@
+Endpoint: class {
+	_host: VectorList<Text>
+	_port: Int
+	host: VectorList<Text> { get {
+		result := VectorList<Text> new()
+		for (i in 0 .. this _host count)
+			result add(this _host[i] take())
+		result
+	}}
+	port ::= this _port
+
+	init: func (=_host, =_port)
+	free: override func {
+		for (i in 0 .. this _host count)
+			this _host[i] free(Owner Receiver)
+		this _host free()
+		super()
+	}
+	toText: func -> Text {
+		result := Text new()
+		for (i in 0 .. this _host count)
+			result += i == 0 ? this _host[i] take() : t"." + this _host[i] take()
+		if (this _port != 0) {
+			portString := this _port toString()
+			result += t":" + Text new(portString)
+			portString free()
+		}
+		result
+	}
+	parse: static func(data: Text) -> This {
+		result: This
+		d := data take()
+		if (!d isEmpty) {
+			splitted := d split(t":")
+			newPort := splitted count > 1 ? splitted remove() toInt() : 0
+			newHost := splitted remove() split(t".")
+			for (i in 0 .. newHost count)
+				newHost[i] = newHost[i] claim()
```

Comment:
I'm still not sure about this. Since the `Text`s you have have `Owner Unknown`, `claim()` will return a copy of the underlying `Buffer`. The original one still remains. So, there should be leaks.